hnn_core/dipole.py

```diff
@@ -52,12 +56,18 @@ def simulate_dipole(net, n_trials=None, record_vsoma=False):
     if n_trials < 1:
         raise ValueError("Invalid number of simulations: %d" % n_trials)
 
-    if record_vsoma is not None and isinstance(record_vsoma, bool):
-        net.params['record_vsoma'] = record_vsoma
+    if isinstance(record_vsoma, bool):
+        net._record_vsoma = record_vsoma
```

Comment:
I'm a bit suspicious if this will work for MPI backend.

Reply:
Really good catch, my local machine was defaulting to the JobLib backend since I didn't specify `n_procs`. MPI produces returns data identical to `record_vsoma=False`. 

Not sure I totally understand the behavior but will dig in tomorrow. Do you have a grasp of the issue? I'm guessing `net._record_vsoma` is not persisting after being passed to the backend.

Reply:
my hunch is that the `MPIBackend` only passes the `params` dictionary not the whole object. I think some effort spent in cleaning up the backends and codepaths would hold us in good stead in the long run.

Reply:
Think I found what you're talking. The params are pickled in `parallel_backends.py` [here](https://github.com/jonescompneurolab/hnn-core/blob/b50f91dcf7c0cdeae87189c9a1dc97f56bddc0b4/hnn_core/parallel_backends.py#L356) and `mpi_child.py` instantiates a new network based solely on the `params` [here](https://github.com/jonescompneurolab/hnn-core/blob/b50f91dcf7c0cdeae87189c9a1dc97f56bddc0b4/hnn_core/mpi_child.py#L103). Short term solution would be reverting back to tacking `record_vsoma` and `record_isoma` onto the `params` dictionary. 

Can you open an issue/PR for a discussion on what needs to be changed for the backends? I don't feel I have a good sense of what's good or bad about the current implementation.

Reply:
Take a look at #142 but you need to muck with the code to realize the limitations/issues. We can't depend on one person for that code. For example, joblib backend allows any picklable object, not just `params` (will open an issue and tag you). Try passing the `record_vsoma` to `_Backend.simulate()` and you'll see how nested and deep the calls quickly get. Shunting through `net.params` is [code smell](https://en.wikipedia.org/wiki/Code_smell)

Reply:
Do you want to pause merging this PR until the parallel issues are addressed? I agree it's not the best solution but I don't see clear workaround besides hardcoding the parameter passing.

Reply:
no, let's move on. Are you able to make it work for both backends though?

Reply:
Yes they both work locally after reverting back to passing through `params`. I've added explicit tests for both backends in `test_dipole.py` (L68). They're a bit over complete as both involve checking the size of the recorded output, let me know if you think it should be cut down.